fedimint-client/src/db.rs

```diff
@@ -642,9 +636,41 @@ pub fn get_core_client_database_migrations()
             })
         }),
     );
+
+    // Fix #7367
+    migrations.insert(
+        DatabaseVersion(3),
+        Box::new(|mut ctx: fedimint_core::db::DbMigrationFnContext<'_, _>| {
+            Box::pin(async move {
+                let mut dbtx = ctx.dbtx();
+
+                for module_id in 0..u16::MAX {
```

Comment:
Prefix query wouldn't work due to the prefix-sharing, so this seems like a reasonable approach. Alternative: use raw prefix querying and length checks before decoding.